crates/driver/src/boundary/settlement.rs

```diff
@@ -326,13 +326,20 @@ fn to_boundary_jit_order(domain: &DomainSeparator, order: &order::Jit) -> Order
         ..Default::default()
     };
     let signature = order.signature.to_boundary_signature();
+    let signer = signature
+        .recover_owner(&order.signature.data.0, domain, &data.hash_struct())
+        .expect("Invalid signer");
 
-    Order {
+    if signer != order.signature.signer.0 {
```

Comment:
I thought of adding this here so we avoid placing a TX which will 100 % fail

Reply:
We definitely need to be able to figure out the actual signer of the order because that is the only way to know whether a JIT order is a CoW AMM order but it should probably be done when converting the JIT `dto` into a domain object.

Reply:
Sure! but the signer is set automatically to the solver address (regardless of the real signer). I can add this check there.

Reply:
We will never place a tx that will fail (we simulate it before). I think this check is a bit weird here. We don't really need to verify the signer here, so it's not clear why it gets asserted (and adds more error handling that may not be required). I'd be in favor of removing it.

Reply:
> but the signer is set automatically to the solver address (regardless of the real signer)

Yeah, this is exactly the part that should change. Instead of always blindly setting the solver address we can just recover the real signer from the signature when we convert the dto instead.

Reply:
Got it! Doing that! :ok_hand: